src/sage/schemes/elliptic_curves/ell_finite_field.py

```diff
@@ -1881,14 +1903,23 @@ def curves_with_j_0(K):
     q = K.cardinality()
     if q % 3 == 2:
         # Then we only have two quadratic twists (and -3 is non-square)
-        return [EllipticCurve(K, [0,a]) for a in [1,-27]]
+        return [EllipticCurve(K, [0, a]) for a in [1, -27]]
     # Now we have genuine sextic twists, find D generating K* mod 6th powers
-    q2 = (q-1)//2
-    q3 = (q-1)//3
-    D = K.gen()
+    q2 = (q - 1) // 2
+    q3 = (q - 1) // 3
+    # Note that computing multiplicative generator is expensive
+    D = K.random_element()
     while not D or D**q2 == 1 or D**q3 == 1:
         D = K.random_element()
-    return [EllipticCurve(K, [0,D**i]) for i in range(6)]
+
+    curves = [EllipticCurve(K, [0, D**i]) for i in range(6)]
+    # TODO (grhkm): issue 37110, Precompute orders of sextic twists + docs
```

Comment:
I don't think it's common (or a good idea) to label TODOs in the code with your name; suggesting to just leave this out.